### ISSUE TITLE
when I change the name of FocalLoss in focal loss.py to FocalLoss2 and modify the config accordingly, the loss drops from  3 to 0.0019

### ISSUE THREAD

**Issue by zhaoguangxiang** [open-mmlab/mmdetection#3650]
Thanks for your error report and we appreciate it a lot.

**Checklist**
1. I have searched related issues but cannot get the expected help.
version is:
 mmdet           2.0.0+unknown
mmcv: 0.5.9

**Describe the bug**
when I change the name of FocalLoss in focal loss.py to FocalLoss2 and modify the config accordingly, the loss drops from  3 to 0.0019.  I did not modify the other code.

**Reproduction**
1. What command or script did you run?
```
save_name=retinanet_r50_fpn_1x_coco_fl_rename
work_dir=work_dirs/${save_name}
PYTHONPATH="tools/..":$PYTHONPATH \
python -m torch.distributed.launch --nproc_per_node=${GPU_NUM} --master_port=$PORT \
    tools/train.py configs/retinanet/${save_name}.py --launcher pytorch \
    --work-dir ${work_dir}
```
2. Did you make any modifications on the code or config? Did you understand what you have modified?
Yes, I modify the name of a criterion and change the config  accordingly, but the results are strange.
3. What dataset did you use?
COCO

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
sys.platform: linux
Python: 3.7.7 (default, May  7 2020, 21:25:33) [GCC 7.3.0]
CUDA available: True
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.1, V10.1.243
GPU 0,1,2,3: GeForce RTX 2080 Ti
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.5.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.6.0a0+82fd1c8
OpenCV: 4.4.0
MMCV: 0.5.9
MMDetection: 2.0.0+unknown
MMDetection Compiler: GCC 7.5
MMDetection CUDA Compiler: 10.1

2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]    conda
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
If applicable, paste the error trackback here.
```
2020-08-29 18:19:13,915 - mmdet - INFO - Distributed training: True
2020-08-29 18:19:14,117 - mmdet - INFO - Config:
model = dict(
    type='RetinaNet',
    pretrained='torchvision://resnet50',
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=True,
        style='pytorch'),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=1,
        add_extra_convs='on_input',
        num_outs=5),
    bbox_head=dict(
        type='RetinaHead',
        num_classes=80,
        in_channels=256,
        stacked_convs=4,
        feat_channels=256,
        anchor_generator=dict(
            type='AnchorGenerator',
            octave_base_scale=4,
            scales_per_octave=3,
            ratios=[0.5, 1.0, 2.0],
            strides=[8, 16, 32, 64, 128]),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoder',
            target_means=[0.0, 0.0, 0.0, 0.0],
            target_stds=[1.0, 1.0, 1.0, 1.0]),
        loss_cls=dict(
            type='FocalLoss2',
            use_sigmoid=True,
            gamma=2.0,
            alpha=0.25,
            loss_weight=1.0),
        loss_bbox=dict(type='L1Loss', loss_weight=1.0)))
train_cfg = dict(
    assigner=dict(
        type='MaxIoUAssigner',
        pos_iou_thr=0.5,
        neg_iou_thr=0.4,
        min_pos_iou=0,
        ignore_iof_thr=-1),
    allowed_border=-1,
    pos_weight=-1,
    debug=False)
test_cfg = dict(
    nms_pre=1000,
    min_bbox_size=0,
    score_thr=0.05,
    nms=dict(type='nms', iou_thr=0.5),
    max_per_img=100)
dataset_type = 'CocoDataset'
data_root = 'data/coco/'


Done (t=0.31s)
creating index...
index created!
2020-08-29 18:20:21,283 - mmdet - INFO - Epoch [1][50/14659]    lr: 0.00099, eta: 2 days, 0:24:19, time: 0.991, data_time: 0.682, memory: 3477, loss_cls: 0.0012, loss_bbox: 0.0007, loss: 0.0019
2020-08-29 18:20:36,820 - mmdet - INFO - Epoch [1][100/14659]   lr: 0.00199, eta: 1 day, 7:46:53, time: 0.311, data_time: 0.006, memory: 3477, loss_cls: 0.0011, loss_bbox: 0.0006, loss: 0.0017

```

**Comment by RyanXLi** [comment]
Hi! The problem might be you need to change [this line](https://github.com/open-mmlab/mmdetection/blob/83f0ca40cc71f6406c7a2346793f38c0dee32593/mmdet/models/losses/__init__.py#L6) correspondingly so that your FocalLoss2 could be correctly registered and built. Please let me know if it solves the problem. Thanks.

**Comment by zhaoguangxiang** [comment]
> Hi! The problem might be you need to change [this line](https://github.com/open-mmlab/mmdetection/blob/83f0ca40cc71f6406c7a2346793f38c0dee32593/mmdet/models/losses/__init__.py#L6) correspondingly so that your FocalLoss2 could be correctly registered and built. Please let me know if it solves the problem. Thanks.

Thanks for your reply, but the related init had been modified. I create the class of FocalLoss2, add it to the focal_loss.py and modified the related config and the init

**Comment by RyanXLi** [comment]
In that case, the problem is probably [here](https://github.com/open-mmlab/mmdetection/blob/ec7d83bb4c2b5ba84f198e5eb5ed491b955bce81/mmdet/models/dense_heads/anchor_head.py#L64). Since the name of the loss is changed to FocalLoss2, the sampling is enabled hence the small loss.

**Comment by zhaoguangxiang** [comment]
> In that case, the problem is probably [here](https://github.com/open-mmlab/mmdetection/blob/ec7d83bb4c2b5ba84f198e5eb5ed491b955bce81/mmdet/models/dense_heads/anchor_head.py#L64). Since the name of the loss is changed to FocalLoss2, the sampling is enabled hence the small loss.

The problem is solved by modifying the code there, btw, what does the sampling mean there?

**Comment by RyanXLi** [comment]
It refers to a positive/negative sampling step typically seen in the Faster-RCNN series of methods. For details, you can have a look at the section 3.1.3 of [the original Faster-RCNN paper](https://arxiv.org/pdf/1506.01497.pdf).

**Comment by zhaoguangxiang** [comment]
> It refers to a positive/negative sampling step typically seen in the Faster-RCNN series of methods. For details, you can have a look at the section 3.1.3 of [the original Faster-RCNN paper](https://arxiv.org/pdf/1506.01497.pdf).

Thank you very much!
